Tests/Bll/Services/ClassServiceTests.cs

```diff
@@ -0,0 +1,240 @@
+                    var result = await _classService.Get(id);
+
+                    // Assert
+                    Assert.NotNull(result);
+                    Assert.IsInstanceOf<ClassDTO>(result);
+                    Assert.AreSame(classDto.Id, result.Id);
+                }
+                */
+        /*
+                [Test]
+                public void Get_WithNonExistingId_ThrowsNotFoundException()
+                {
+                    // Arrange
+                    var id = 1;
+
+                    _contextMock.Setup(c => c.Classes.Include(c => c.Teacher).Include(c => c.Group).Where(c => c.Id == id))
+                        .Returns(MockDbSet(new List<Class> { }));
+
+                    // Act & Assert
+                    Assert.ThrowsAsync<NotFoundException>(() => _classService.Get(id));
+                }
+                */
+        /*
+                [Test]
+                public async Task GetClassesForUserOnDay_WithValidData_ReturnsListOfClassDTOs()
+                {
+                    // Arrange
+                    var user = new User();
+                    var date = new DateOnly(2023, 5, 19);
+                    var existingClasses = new List<Class> { new Class(), new Class() };
+
+                    _contextMock.Setup(c => c.Classes.Include(c => c.Teacher).Include(c => c.Group).ThenInclude(g => g.Users).Where(c => c.StartDateTime.Year == date.Year && c.StartDateTime.Month == date.Month && c.StartDateTime.Day == date.Day))
+                        .Returns(MockDbSet(existingClasses));
+                    _mapperMock.Setup(m => m.Map<List<ClassDTO>>(existingClasses))
+                        .Returns(existingClasses.Select(c => new ClassDTO(c.Id, c.Name, c.StartDateTime, new UserInfoDTO(c.Teacher.Id, c.Teacher?.UserName ?? "Name"), new GroupDTO(c.Group.Id, c.Group.Name))).ToList());
+
+                    // Act
+                    var result = await _classService.GetClassesForUserOnDay(user, date);
+
+                    // Assert
+                    Assert.NotNull(result);
+                    Assert.IsInstanceOf<IEnumerable<ClassDTO>>(result);
+                    Assert.AreEqual(existingClasses.Count, result.Count());
+                }
+        */
+        [Test]
+        public async Task Update_WithExistingDto_UpdatesClass()
+        {
+            // Arrange
+            var existingClass = new Class();
+            var dto = new ClassDTO(1, "ClassName", new DateTime(2020, 1, 1, 12, 0, 0), new UserInfoDTO(1, "Teacher"), new GroupDTO(1, "Group"));
+
+            _contextMock.Setup(c => c.Classes.Find(dto.Id))
+                .Returns(existingClass);
+            _mapperMock.Setup(m => m.Map(dto, existingClass));
+            _contextMock.Setup(c => c.SaveChangesAsync(default))
+                .ReturnsAsync(1);
+
+            // Act
+            await _classService.Update(dto);
+
+            // Assert
+            _contextMock.Verify(c => c.SaveChangesAsync(default), Times.Once);
```

Comment:
Talán érdemes lehet még itt azt is ellenőrizni, hogy az értékek a dto-ban és a példányban megegyeznek-e.